pkg/cmd/gist/view/view.go

```diff
@@ -115,17 +126,32 @@ func viewRun(opts *ViewOptions) error {
 			_, err := fmt.Fprint(opts.IO.Out, "\n")
 			return err
 		}
+
 		return nil
 	}
 
 	if opts.Filename != "" {
 		gistFile, ok := gist.Files[opts.Filename]
+		isBinary, err := shared.DetectedBinaryFileContents(gistFile.Content, true)
+		if err != nil {
+			return err
+		}
+
+		if isBinary {
+			return fmt.Errorf("Error: file contents is binary")
+		}
+
 		if !ok {
 			return fmt.Errorf("gist has no such file: %q", opts.Filename)
 		}
 		return render(gistFile)
 	}
 
+	if err := opts.IO.StartPager(); err != nil {
```

Comment:
This change made it so that the pager isn't used when rendering a single file. Was this intentional?

Reply:
If you're referring to the `startPager`'s block coming down from above, it was because, irrespective of the file contents, the pager was initialized earlier. And yes, if a gist has only one file and it is a binary file, I thought there was no need to initialize the pager itself.